pages/api/v1/progressReport/editProgressReportSummary.ts

```diff
@@ -55,34 +55,39 @@ const handler: NextApiHandler<{ message: string; progress_report?: object }> = a
 
 		const existingPost = postDoc.data() as Post;
 
-		if (!existingPost.progress_report) {
-			return res.status(400).json({ message: 'No progress report found for the specified post.' });
+		if (!existingPost.progress_report || !Array.isArray(existingPost.progress_report)) {
+			return res.status(400).json({ message: 'No progress reports found for the specified post.' });
 		}
 
-		const updatedProgressReport = {
-			...existingPost.progress_report,
-			isEdited: true,
-			progress_summary: summary
-		};
+		const updatedProgressReports = existingPost.progress_report.map((report) => {
+			if (report.id === reportId) {
+				return {
+					...report,
```

Comment:
Please use `created_at.toDate()` for `report.created_at` otherwise it will save the firebase timestamp in different format